src/sagemaker/utils.py

```diff
@@ -538,6 +538,24 @@ def get_ecr_image_uri_prefix(account, region):
     return "{}.dkr.ecr.{}.{}".format(account, region, domain)
 
 
+def sts_regional_endpoint(region):
+    """Get the AWS STS endpoint specific for the given region.
+
+    We need this function because the AWS SDK does not yet honor
+    the ``region_name`` parameter when creating an AWS STS client.
+
+    For the list of regional endpoints, see
+    https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html#id_credentials_region-endpoints.
+
+    Args:
+        region (str): AWS region name
+
+    Returns:
+        str: AWS STS regional endpoint
+    """
+    return "sts.{}.amazonaws.com".format(region)
```

Comment:
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html#Region_Endpoints
We'll need some specific logic for ap-east-1 and me-south-1
:(